lib/dal/src/attribute/value.rs

```diff
@@ -719,40 +719,38 @@ impl AttributeValue {
         };
 
         let component_id = Self::component_id(ctx, input_attribute_value_id).await?;
-        let input_socket_match = InputSocketMatch {
+
+        let component_input_socket = ComponentInputSocket {
             component_id,
             input_socket_id,
             attribute_value_id: input_attribute_value_id,
         };
         let mut inputs = vec![];
 
-        match Component::find_available_inferred_connections_to_input_socket(
-            ctx,
-            input_socket_match,
-        )
-        .await
-        {
-            Ok(maybe_matches) => {
-                for output_match in maybe_matches {
+        match ComponentInputSocket::find_inferred_connections(ctx, component_input_socket).await {
+            Ok(connections) => {
+                for component_output_socket in connections {
                     // Both deleted and non deleted components can feed data into deleted components.
                     // ** ONLY ** non-deleted components can feed data into non-deleted components
-                    if !Component::should_data_flow_between_components(
+                    if Component::should_data_flow_between_components(
                         ctx,
-                        input_socket_match.component_id,
-                        output_match.component_id,
+                        component_input_socket.component_id,
+                        component_output_socket.component_id,
                     )
                     .await
                     .map_err(Box::new)?
                     {
-                        return Ok(vec![]);
+                        // XXX: We need to properly handle the difference between "there is
```

Comment:
Is this check no longer necessary?

Reply:
no it is, I thought about encapsulating that, but we need to return the connections even if data shouldn't flow (because we show them on the UI)